scripts/DallasFuncs.h

```diff
@@ -0,0 +1,360 @@
+#pragma once
+#include <variant>
+#include <initializer_list>
+#include <vector>
+
+class user_var : public std::variant<float, int32_t> {
+	private:
+	using VarT = std::variant<float, int32_t>;
+
+	public:
+	// For compatibility with existing code, override op= with one that
+	// retains the active type, requiring instead set_type<> to change it.
+	template<typename T> void operator=(T &&v) noexcept {
+		VarT &self = *this;
+		if (std::get_if<float>(this))
+			self = static_cast<float>(v);
+		else
+			self = static_cast<int32_t>(v);
+	}
+	template<typename T> void set_type() noexcept { static_cast<VarT &>(*this) = T{}; }
+	void operator++(int) noexcept { std::visit([](auto &&uv) { ++uv; }, *this); }
```

Comment:
This is causing compilation problems on GCC 9.4 (Ubuntu 20.04), because [p2162r2](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p2162r2.html) is not implemented. It has only been backported [to GCC 11](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=90943). We'll need to implement our own `visit` function to circumvent that

Reply:
@jengelh could you please take a look at https://github.com/DescentDevelopers/Descent3/pull/592 ?